### PR TITLE
AssertJ Use assertJ in all tests

### PR DESCRIPTION
**Pull request by D00ktora**
Added tests with assertThat in every test case
and fullfilled the requirement of
assertThatThrownBy in test where an exeption is thrown.

### CODE REVIEW

**Review by burberius — APPROVED** [review]

File: src/test/java/project/web/GasStationControllerTest.java [hunk 1]
```diff
@@ -46,11 +46,10 @@ void getGasStationByNameTestWithInvalidInput() {
         String stationName = "Non-existent Station";
         when(gasStationService.getStationsByName(stationName)).thenReturn(new ArrayList<>());
 
-        //WHEN
-        ResponseEntity<List<GasStationDTO>> response = gasStationController.getGasStationByName(stationName);
-
-        //THEN
-        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+        // WHEN THEN
+        assertThat(gasStationController.getGasStationByName(stationName))
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: i would keep the WHEN separated from the THEN

File: src/test/java/project/web/GasStationControllerTest.java [hunk 2]
```diff
@@ -60,12 +59,13 @@ void getGasPriceInfoTestWithValidInfo() {
         GasPriceInfoDTO gasPriceInfo = createTestGasPriceInfoDTO();
         when(gasStationService.getGasPriceInfo(fuelType)).thenReturn(gasPriceInfo);
 
-        // WHEN
-        ResponseEntity<GasPriceInfoDTO> response = gasStationController.getGasPriceInfo(fuelType);
-
-        // THEN
-        assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals(gasPriceInfo, response.getBody());
+        // WHEN THEN
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: here great example, why you should keep the WHEN separated - in this case you have duplications!

File: src/test/java/project/web/GasStationControllerTest.java [hunk 3]
```diff
@@ -74,11 +74,10 @@ void getGasPriceInfoTestWithInvalidInput() {
         String fuelType = "Diesel";
         when(gasStationService.getGasPriceInfo(fuelType)).thenReturn(null);
 
-        // WHEN
-        ResponseEntity<GasPriceInfoDTO> response = gasStationController.getGasPriceInfo(fuelType);
-
-        // THEN
-        assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+        // WHEN THEN
+        assertThat(gasStationController.getGasPriceInfo(fuelType))
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: same here, please do not do it if possible

File: src/test/java/project/web/GasStationControllerTest.java [hunk 4]
```diff
@@ -32,12 +31,13 @@ void getGasStationByNameTestWithValidInput() {
         String stationName = "Test Station";
         List<GasStationDTO> stations = setupResultsFromGetStationByNameToReturnListWithOneStation(stationName);
 
-        // WHEN
-        ResponseEntity<List<GasStationDTO>> response = gasStationController.getGasStationByName(stationName);
-
-        // THEN
-        assertEquals(HttpStatus.OK, response.getStatusCode());
-        assertEquals(stations, response.getBody());
+        // WHEN THEN
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: here too - duplications

File: src/test/java/project/service/GasStationServiceImplTest.java [hunk 5]
```diff
@@ -82,28 +80,34 @@ void getGasPriceInfoTestWithThreeStations(String fuelType) {
...
+        setupFunctionalityOfFindAllByFuelTypeWithTwoGasStations(fuelType);
 
-        // WHEN
-        GasPriceInfoDTO gasPriceInfo = gasStationService.getGasPriceInfo(text);
-
-        // THEN
-        assertEquals(BigDecimal.valueOf(1.5).setScale(2, RoundingMode.HALF_UP), gasPriceInfo.getMedian());
-        assertEquals(BigDecimal.valueOf(1.0), gasPriceInfo.getMin());
-        assertEquals(BigDecimal.valueOf(2.0), gasPriceInfo.getMax());
+        // WHEN THEN
```
Comment (cupakob) [inline src/test/java/project/service/GasStationServiceImplTest.java]: here three calls for the same - `gasStationService.getGasPriceInfo()`

File: src/test/java/project/web/GasStationControllerTest.java (hunk 1 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: It’s fixed and split.

File: src/test/java/project/web/GasStationControllerTest.java (hunk 2 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: fixed

File: src/test/java/project/web/GasStationControllerTest.java (hunk 3 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: fixed

File: src/test/java/project/web/GasStationControllerTest.java (hunk 4 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: fixed

File: src/test/java/project/service/GasStationServiceImplTest.java (hunk 5 above)
Comment (D00ktora) [reply src/test/java/project/service/GasStationServiceImplTest.java]: extracted and fixed

File: src/test/java/project/web/GasStationControllerTest.java [hunk 6]
```diff
@@ -33,11 +33,15 @@ void getGasStationByNameTestWithValidInput() {
         List<GasStationDTO> stations = setupResultsFromGetStationByNameToReturnListWithOneStation(stationName);
 
         // WHEN
-        ResponseEntity<List<GasStationDTO>> response = gasStationController.getGasStationByName(stationName);
+        ResponseEntity<List<GasStationDTO>> gasStationByName = gasStationController.getGasStationByName(stationName);
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: jfi: i think, the naming here can be better, imho "ByName" is just too much. Why i need to know, how you got this object?  Also variables with `DTO` as suffix are imho not really great.

File: src/test/java/project/web/GasStationControllerTest.java [hunk 7]
```diff
@@ -75,10 +85,12 @@ void getGasPriceInfoTestWithInvalidInput() {
         when(gasStationService.getGasPriceInfo(fuelType)).thenReturn(null);
 
         // WHEN
-        ResponseEntity<GasPriceInfoDTO> response = gasStationController.getGasPriceInfo(fuelType);
+        ResponseEntity<GasPriceInfoDTO> gasPriceInfo = gasStationController.getGasPriceInfo(fuelType);
```
Comment (cupakob) [inline src/test/java/project/web/GasStationControllerTest.java]: here the naming is much better

File: src/test/java/project/web/GasStationControllerTest.java (hunk 6 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: accepted and fixed :)

File: src/test/java/project/web/GasStationControllerTest.java (hunk 7 above)
Comment (D00ktora) [reply src/test/java/project/web/GasStationControllerTest.java]: here i remove Info and leave it as gasPrice.

File: src/test/java/project/service/GasStationServiceImplTest.java [hunk 8]
```diff
@@ -81,55 +79,71 @@ void verifyOnlyOneCallOfFindByNameInGetStationByName() {
     void getGasPriceInfoTestWithThreeStations(String fuelType) {
         // GIVEN
         setupFunctionalityOfFindAllByFuelTypeWithThreeGasStations(fuelType);
-
+
```
Comment (cupakob) [inline src/test/java/project/service/GasStationServiceImplTest.java:82]: please revert lines 82, 124 and 139, this changes are not really needed

**Pull request merged**